### ISSUE TITLE
Cat does not escape newline characters before printing.

### ISSUE THREAD

**Issue by neruok** [xonsh/xonsh#2565]
I found that cat doesn't work like the GNU Coreutils cat does. It outputs differently when escaped characters are part of a file and interprets those characters instead of ignoring them.

Below I've put a log showing the difference in output.
```
neruok@cody-ubuntu ~/Workspace/test/xonsh $ ls
cat.txt
neruok@cody-ubuntu ~/Workspace/test/xonsh $ bash -c 'cat cat.txt'
\n\n
\n\n
neruok@cody-ubuntu ~/Workspace/test/xonsh $ $(bash -c 'cat cat.txt')
'\\n\\n\n\\n\\n\n'
neruok@cody-ubuntu ~/Workspace/test/xonsh $ cat cat.txt






neruok@cody-ubuntu ~/Workspace/test/xonsh $ $(cat cat.txt)
'\n\n\n\n\n\n'
```

```
neruok@cody-ubuntu ~/Workspace/test/xonsh $ xonfig
+------------------+---------------------+
| xonsh            | 0.5.12              |
| Git SHA          | 8941dfdd            |
| Commit Date      | Jun 7 16:04:39 2017 |
| Python           | 3.5.2               |
| PLY              | 3.9                 |
| have readline    | True                |
| prompt toolkit   | 1.0.15              |
| shell type       | prompt_toolkit      |
| pygments         | 2.2.0               |
| on posix         | True                |
| on linux         | True                |
| distro           | Ubuntu              |
| on darwin        | False               |
| on windows       | False               |
| on cygwin        | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+

```

I'm using an older configuration, though it seems that this issue should still be present given that line 40 of [this file](https://github.com/xonsh/xonsh/blob/master/xonsh/xoreutils/cat.py#L40) is the culprit. Changing
```python
print(_r.decode('unicode_escape'), flush=True, file=out)
```
to
```python
print(_r.decode('utf8'), flush=True, file=out)
```
fixes the issue.

**Comment by scopatz** [comment]
Thanks for the report @neruok! The fix should probably be to decode as `$XONSH_ENCODING`. Would you mind putting in a PR to fix this, please? Thanks!

**Comment by scopatz** [comment]
Or anyone is welcome to :smile:

**Comment by gforsyth** [comment]
Resolved in #2569
